docs: add concept for DNS proxying in Istio (#1886)
kyma-project/istio: docs/user/00-35-dns-proxying.md

In some cases TCP traffic in Istio is routed based on destination IP and port only. Unlike HTTP traffic, which includes a Host header, TCP has no additional metadata for routing decisions.

For example when multiple external TCP services share the same port (for example, two databases on port 3306) and they don't have stable IP addresses so the specified ServiceEntries has resolution set to `DNS` Istio cannot distinguish between them. The proxy creates a single listener on `0.0.0.0:{port}` and forwards traffic to only one destination (external TCP service).

DNS proxying solves this by auto-allocating virtual IPs (VIPs) from the `240.240.0.0/16` range to each ServiceEntry. This gives each external TCP service a unique address, enabling the proxy to route traffic correctly even when sharing the same port.

## How to Enable DNS Proxying

DNS proxying is disabled by default. You can enable it globally for the entire mesh or locally per workload. Local per-workload configuration overrides the global mesh configuration.

### Global Mesh Configuration

You can enable DNS proxying globally using the Kyma Dashboard or kubectl.

#### Using Kyma Dashboard

> 1. Navigate to **Cluster Details** in the Kyma Dashboard

Reviewer: ```suggestion
1. Go to **Cluster Details** in Kyma dashboard.
```

> 2. In the `kyma-system` namespace go to **Istio** section

Reviewer: ```suggestion
2. In the `kyma-system` namespace, go to the **Istio** section.
```

3. Click **Edit**
4. In the configuration, set **Enable global DNS Proxying** to `true`
5. Save the changes

#### Using kubectl

Set **enableDNSProxying** to `true` in the Istio custom resource (CR):

```yaml
apiVersion: operator.kyma-project.io/v1alpha2
kind: Istio
metadata:
  name: default
  namespace: kyma-system
spec:
  config:
    enableDNSProxying: true
```

You can also use `kubectl patch`:

```bash
kubectl patch istio default -n kyma-system --type=merge -p '{"spec":{"config":{"enableDNSProxying":true}}}'
```

> 

Reviewer: ```suggestion
<!-- tabs:end -->

```

To verify the configuration:

```bash
kubectl get istio default -n kyma-system -o jsonpath='{.spec.config.enableDNSProxying}'
```

This enables DNS proxying globally for all proxies in the mesh.

### Local Per-Workload Configuration

Add the `proxy.istio.io/config` annotation to enable DNS proxying for a specific workload:

```yaml
apiVersion: apps/v1
kind: Deployment
metadata:
  name: my-app
spec:
  template:
    metadata:
      annotations:
        proxy.istio.io/config: |
          proxyMetadata:
            ISTIO_META_DNS_CAPTURE: "true"
    spec:
      containers:
      - name: my-app
        image: my-app:latest
```

> You can also use `kubectl` to add the annotation to an existing Deployment:

Reviewer: ```suggestion
You can also use kubectl to add the annotation to an existing Deployment:
```
